### PR TITLE
Bug fix for filter commands with list of conditions

### PR DESCRIPTION
**Pull request by jigold**
- List of str inputs for condition are modified such that each condition is wrapped in parentheses and multiple conditions are separated by `&&`. Previous was conditions were comma separated.

### CODE REVIEW

**Comment by tpoterba** [comment]
I'm now unsure this is the right thing to do.  Should these two semantics be equivalent?
```
vds.filter_variants_expr(['cond1', 'cond2', 'cond3', ...])
```

```
(vds.filter_variants_expr('cond1')
    .filter_variants_expr('cond2')
    .filter_variants_expr('cond3')
    ...)
```

If so, we don't always want to join with `&&`.  If `remove=True`, we should join with `||`

**Comment by cseed** [comment]
Yeah, I'm not sure I like this change, for exactly the reason @tpoterba mentions: should we join with `&&` or `||`?  This is more natural with files since the only thing a list can mean is to load them all (which is already supported in the syntax).

What was the particular use case driving this?

**Comment by jigold** [comment]
The use case is filter commands right now can take a list of expressions. Seems like it's easier to not allow list of str as an input.

**Comment by cseed** [comment]
> filter commands right now can take a list of expressions

What do you mean?  What filter commands?  Like `filtervariants expr`?  It only takes a single expression.

**Comment by tpoterba** [comment]
Jackie submitted this PR to fix a problem in the master that lets you pass `list(str)` to filter commands, which just joins them.  

One solution was to join with `&`, but we later realized this had issues.  I think the better fix is to just require that the filter commands take `str`

**Comment by jigold** [comment]
`filter_samples_expr` and `filter_variants_expr` both take a list of str as input.

**Comment by cseed** [comment]
Ah!  That's a mistake.  The correct fix for this is to make them just take a str.

**Comment by jigold** [comment]
back to you

**Comment by cseed** [comment]
I wouldn't do this `&&` business in any of them.  They should each take a single condition.

**Comment by jigold** [comment]
@cseed I fixed the remainder. Sorry about that.

File: python/pyhail/tests.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def test_dataset(self):
         
         sample2_split.export_gen('/tmp/sample2.gen')
 
-        (sample2.filter_genotypes('g.isHet && g.gq > 20')
+        (sample2.filter_genotypes(['g.isHet', 'g.gq > 20'])
```
Comment (cseed) [inline python/pyhail/tests.py]: I think this will break now.  Otherwise, looks good.

**Pull request merged**